Added Update-VSTeamGitRepositoryDefaultBranch

rbleattler opened the pull request:
# PR Summary

- Added Update-VSTeamGitRepositoryDefaultBranch to allow for changing the default branch of a repository
- Fixed bad link in README.md
- Bumped manifest version
- Cleaned up comments/debug messages
- Proposed changes from original PR : MethodsAndPractices/vsteam/pull/474

## PR Checklist

- [X] [Write Help](https://github.com/DarqueWarrior/vsteam/blob/master/.github/CONTRIBUTING.md#write-help)
- [X] [Write Unit Test](https://github.com/DarqueWarrior/vsteam/blob/master/.github/CONTRIBUTING.md#write-unit-test)
- [X] [Update CHANGELOG.md](https://github.com/DarqueWarrior/vsteam/blob/master/.github/CONTRIBUTING.md#update-changelogmd)

Commit:
- Added Update-VSTeamGitRepositoryDefaultBranch to allow for changing the default branch of a repository.
- Fixed bad link in README.md

Commit:
Clean up comments, and fixed/updated urls

Commit:
Added logic to only try to import vstem-lib.dll if exists and is not imported.

Commit:
Added fake Get-VSTeamGitRepository call (since pester doesn't recognize it as something when mocked)
Removed call to dot source common.ps1
Removed debug messages
Cleaned up comments
Bumped Version
Updated Changelog

rbleattler committed:
Merge pull request #1 from MethodsAndPractices/trunk

fix: Remove space in -Build paramter of example 4 (#472)

rbleattler commented:
@SebastianSchuetze do I need to re-submit with SupportsShouldProcess implemented?

SebastianSchuetze commented:
What do you mean resubmit? You can just keep editing your branch until everything is fine. You didn't need to close the PR and reopen one.

Just commit here until everything is fine. At the end we do a squash merge anyways.

rbleattler commented:
Thanks, I learn something new every day. I'll try to get this updated on Monday.

Commit:
- Fixed default branch name check.
- Added logic to pull refs and validate branch name
- Updated call to use splatting for readability

Commit:
- Added mock for Get-VSTeamGitRef
- Added test for non-existent branch name

Commit:
- Added sample file for gitref call in Update-VSTeamGitRepositoryDefaultBranch

rbleattler commented:
Updated Update-GitVSTeamGitRepositoryDefaultBranch with SupportsShouldProcess / -WhatIf support.

rbleattler commented:
I believe everything that needed to be fixed has been fixed, including adding support for SupportsShouldProcess / -WhatIf

SebastianSchuetze commented on Tests/function/tests/_testInitialize.ps1:
```diff
@@ -11,8 +11,18 @@ Import-Module SHiPS
-Add-Type -Path "$baseFolder/dist/bin/vsteam-lib.dll"
+
+# Changed to only import this type if the dll is missing and exists in the path, since PS throws an exception when trying to import a dll already in memory
+$vsTeamDllPath = "$baseFolder/dist/bin/vsteam-lib.dll"
```
thanks for this addition! :)

SebastianSchuetze committed:
removed some line in changelog

SebastianSchuetze commented:
I removed some lines in the changelog because it is not interesting if there are typo changes. Only real changes for somebody using the modules is interesting. E.g. bug fixes or new features in cmdlets or the documentation of the cmdlets themselve.

approved and merged. Thanks very much!!
